lib/trento/activity_logging/parser/metadata_enricher.ex

```diff
@@ -79,6 +80,7 @@ defmodule Trento.ActivityLog.Logger.Parser.MetadataEnricher do
   defp detect_enrichment(:cluster, {_, %{cluster_id: id}}), do: {:ok, id}
   defp detect_enrichment(:database, {_, %{database_id: id}}), do: {:ok, id}
   defp detect_enrichment(:sap_system, {_, %{sap_system_id: id}}), do: {:ok, id}
+  defp detect_enrichment(:user, {_, %{user_id: id}}), do: {:ok, id}
```

Comment:
In the end, I found out this works:
```elixir
 defp detect_enrichment(
         target_entity,
         {_,
          %{
            resource_id: resource_id,
            resource_type: target_entity
          }}
       ),
       do: {:ok, resource_id}
```

Also, I moved it at the bottom being less specific than the others

Reply:
Do you think it's necessary? What's wrong with having (for example) a cluster enriched regardless of the activity type?

Reply:
Not strictly necessary perhaps and there is nothing wrong in enriching a cluster (or anything else) for different activities.

For example it happens for cluster checks execution being requested, cluster tagging/untagging and for any domain event carrying a cluster id but not the cluster name.

There is a slightly difference in the semantics, imho.

For instance the following two clauses could be merged in one, however we'd miss a bit of context about connection vs domain event activities.
```elixir
# connection related
defp detect_enrichment(
         :cluster,
         {:cluster_checks_execution_request, %{cluster_id: cluster_id}}
       ),
       do: {:ok, cluster_id}

# domain event related
defp detect_enrichment(:cluster, {_, %{cluster_id: id}}), do: {:ok, id}
```
Now that you make me think a couple of comments would be helpful 😄

Reply:
Same goes for 
```elixir
  defp detect_enrichment(:host, {:host_checks_execution_request, %{host_id: host_id}}),
    do: {:ok, host_id}
```
it can be avoided.

Reply:
I am fine either way, as long as we are clear on the two natures of activities (connection related and domain event related) and nothing breaks 😄